src/Manager.hs

```diff
@@ -36,13 +36,40 @@ type ManagerSig sig m = ( Has (State FrameTable) sig m
                         , Has (Catch ManagerError) sig m
                         )
 
+
 -- Find free RAM frame (if available), if not, unload frame from RAM to SWAP and use it
-allocPage :: ManagerSig sig m => ProcessId -> m (Maybe (Page 'Ram))
-allocPage pid = undefined
+allocPage :: ManagerSig sig m => ProcessId -> m (Page 'Ram)
+allocPage pid = do
+  -- get offset of any free frame
+  offM <- getFreeFrameOffset @'Ram
+  case offM of
+    -- in case free frame exists, then alloc a new page
+    Just off -> do
+      np <- createPage (offToId off) pid
+      return np
+    -- otherwise move specific page to swap and alloc a new page
+    Nothing -> do
+      -- unload a page to swap and create a new one instead
+      p@(Page {frId}) <- findPageToUnload
```

Comment:
The convention is to place spaces between curly brackets when using field puns
Like this: `Page { frId }`